src/common/os/posix/os_utils.cpp

```diff
@@ -393,4 +395,147 @@ void CtrlCHandler::handler(void*)
 	terminated = true;
 }
 
+namespace {
+
+	class Converter
+	{
+	public:
+		virtual char* convert(const char* fromBuf, size_t fromLen) = 0;
+		virtual void destroy(char* &outBuf) = 0;
+	};
+
+	// No real conversion, system locale is UTF-8 or iconv is not available
+	class NullConverter : public Converter
+	{
+		char* convert(const char* fromBuf, size_t fromLen) override
+		{
+			return const_cast<char*>(fromBuf);
+		}
+		void destroy(char* &outBuf)
+		{
+			outBuf = NULL;
+		}
+	};
+
+#ifdef HAVE_ICONV_H
+	class IConverter : public Converter
+	{
+	public:
+
+		IConverter(const char* sysLocale) : ic(NULL)
```

Comment:
Then it should better be renamed to IntlConverter (or whatever "I" stands for there).